koku/sources/sources_http_client.py

```diff
@@ -221,20 +218,9 @@ def _get_aws_credentials(self, _):
         if not auth_data:
             raise SourcesHTTPClientError(f"Unable to get AWS roleARN for Source: {self._source_id}")
 
-        # Platform sources is moving the ARN from the password to the username field.
-        # We are supporting both until the this change has made it to all environments.
         username = auth_data.get("username")
         if username:
-            return {"role_arn": username}
-
-        auth_id = auth_data.get("id")
-        auth_internal_url = (
-            f"{self._internal_url}/{ENDPOINT_AUTHENTICATIONS}/{auth_id}?expose_encrypted_attribute[]=password"
-        )
-        auth_internal_response = self._get_network_response(auth_internal_url, "Unable to get AWS RoleARN")
-        password = auth_internal_response.get("password")
-        if password:
-            return {"role_arn": password}
+            return {"role_arn": username, "external_id": auth_data.get("extra", {}).get("external_id")}
```

Comment:
Is it ok for `external_id` to be `None`?

Reply:
for now, yes.